### PR TITLE
Add min deposit amount for currencies and validation for small deposits

### LINKED ISSUE #1743: Deposits are published to RabbiMQ before save
```
Received: {"id":4,"locale":"en"}
Received request for deposit collection id: 4.
  ^[[1m^[[36mDeposit Load (0.7ms)^[[0m  ^[[1mSELECT  `deposits`.* FROM `deposits` WHERE `deposits`.`id` = 4 LIMIT 1^[[0m
The deposit with id: 4 doesn't exist.
```

### LINKED ISSUE #1744: Tiny deposits should be skipped and we shouldn't collect them

### LINKED ISSUE #1754: Avoid duplicate of deposits after execute in deposit collection daemon

### PR DESCRIPTION
**Pull request by mnaichuk**
closes #1743 
closes #1744 
closes #1754

### CODE REVIEW

**Review by ysv — CHANGES_REQUESTED** [review]
1. Add validation to Deposit model
2. Add new field to currencies.yml.erb

**Review by ysv — APPROVED** [review]

File: spec/models/deposit_spec.rb [hunk 1]
```diff
@@ -27,7 +27,7 @@
     let(:deposit) { build(:deposit_usd, member: member, amount: amount, currency: currency) }
     it 'fails validation' do
       expect(deposit.save).to eq false
-      expect(deposit.errors.full_messages).to eq ['Amount must be greater than 0.0']
```
Comment (ysv) [inline spec/models/deposit_spec.rb:30]: Lets use single quotes

File: db/migrate/20181004114428_add_min_deposit_amount_to_currencies.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddMinDepositAmountToCurrencies < ActiveRecord::Migration
+  def change
+    add_column :currencies, :min_deposit_amount, :decimal, precision: 32, scale: 16, default: 0.0, null: false, after: 'quick_withdraw_limit'
```
Comment (ysv) [inline db/migrate/20181004114428_add_min_deposit_amount_to_currencies.rb]: use symbols here `after: :quick_withdraw_limit`

File: app/services/blockchain_service/dash.rb [hunk 3]
```diff
@@ -20,6 +20,11 @@ def build_deposits(block_json, block_id)
           deposit_txs = client.build_transaction(txn, block_id, payment_address.address)
 
           deposit_txs.fetch(:entries).each_with_index do |entry, i|
+            if entry[:amount] <= payment_address.currency.min_deposit_amount
```
Comment (ysv) [inline app/services/blockchain_service/dash.rb:23]: make this update everywhere
```ruby
# If deposit amount smaller then currency min deposit amount.
if entry[:amount] <= payment_address.currency.min_deposit_amount
  # Currently we just skip small deposits. Custom behavior will be implemented later.
next
end
```

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/services/blockchain_service/ripple.rb [hunk 4]
```diff
@@ -48,6 +48,10 @@ def build_deposits(transactions, ledger_index)
         payment_addresses_where(address: address) do |payment_address|
           deposit_txs = client.build_transaction(tx: tx, currency: payment_address.currency)
           deposit_txs.fetch(:entries).each_with_index do |entry, index|
+            if entry[:amount] <= payment_address.currency.min_deposit_amount
+              # Currently we just skip small deposits. Custom behavior will be implemented later.
```
Comment (ysv) [inline app/services/blockchain_service/ripple.rb:52]: Show message in logs

**Review by ysv — CHANGES_REQUESTED** [review]

**Comment by huuhait** [comment]
can you add to gon?

**Pull request merged**